### ISSUE TITLE
While trying to soar while flying, that soar button trigger this following error.

### ISSUE THREAD

**Issue by BigFire321** [kaelonR/WoW-BuyemAll-live#13]
I had just completed a pet battle prior to to flying.  Pressing soar cause the following error.  Trid again with just the following mod: bug grabber, rematch, pet battle scripts and buyemall and was not able to recreate the issue.

1x [ADDON_ACTION_FORBIDDEN] AddOn 'BuyEmAll' tried to call the protected function 'UseAction()'.
[string "@!BugGrabber/BugGrabber.lua"]:485: in function <!BugGrabber/BugGrabber.lua:485>
[string "=[C]"]: in function `UseAction'
[string "@Blizzard_FrameXML/Mainline/SecureTemplates.lua"]:342: in function `handler'
[string "@Blizzard_FrameXML/Mainline/SecureTemplates.lua"]:667: in function <...ddOns/Blizzard_FrameXML/Mainline/SecureTemplates.lua:649>
[string "@Blizzard_FrameXML/Mainline/SecureTemplates.lua"]:681: in function <...ddOns/Blizzard_FrameXML/Mainline/SecureTemplates.lua:674>
[string "@Blizzard_FrameXML/Mainline/SecureTemplates.lua"]:723: in function `SecureActionButton_OnClick'
[string "@Blizzard_ActionBar/Mainline/ActionButton.lua"]:100: in function `TryUseActionButton'
[string "@Blizzard_ActionBar/Mainline/ActionButton.lua"]:135: in function `ActionButtonDown'
[string "ACTIONBUTTON2"]:2: in function <[string "ACTIONBUTTON2"]:1>

Locals:
_ = Frame {
}
event = "ADDON_ACTION_FORBIDDEN"
events = <table> {
}

**Comment by kaelonR** [comment]
Hey, I have a suspicion that this is Blizzard's errors being weird again. BuyEmAll does not do anything with action buttons, and doesn't change anything in the default UI, which make sme think it's probably another addon that is causing this issue.

Can you consistently reproduce this with a pet battle?
If yes, can you try disabling all your addons **except** for BuyEmAll, and then see if the issue persists?

**Comment by zcourge** [comment]
I too get the same problems since today, hasn't been happening earlier

```
1x [ADDON_ACTION_BLOCKED] AddOn 'BuyEmAll' tried to call the protected function 'MultiBarBottomRightButton5:SetAttribute()'.
[string "@!BugGrabber/BugGrabber.lua"]:485: in function <!BugGrabber/BugGrabber.lua:485>
[string "=[C]"]: in function `SetAttribute'
[string "@Blizzard_ActionBar/Mainline/ActionButton.lua"]:488: in function `UpdatePressAndHoldAction'
[string "@Blizzard_ActionBar/Mainline/ActionButton.lua"]:554: in function `Update'
[string "@Blizzard_ActionBar/Mainline/ActionButton.lua"]:1078: in function `OnEvent'
[string "@Blizzard_ActionBar/Mainline/ActionButton.lua"]:302: in function <Blizzard_ActionBar/Mainline/ActionButton.lua:279>

Locals:
_ = Frame {
}
event = "ADDON_ACTION_BLOCKED"
events = <table> {
}
```

**Comment by raysmith59** [comment]
I noticed this earlier but didn't have a chance to report it until now, when I change talent loadouts I get this error anytime I use hotkeys; clicking the buttons is fine, it's just hotkeys.  Doing a reload resolves the issue but I now have to do that every time I change talent loadouts.

**Comment by raysmith59** [comment]
I'm getting the same error as OP and another one I'll post separately although I don't know what caused that one, I didn't realize it was different until I just looked back.  Happens every time I change talent loadouts and then use any hotkey afterward.  Reloading resolves it until the next loadout change so not the end of the world.  I likely won't have time to diagnose further until after raid but will follow up later.

2x [ADDON_ACTION_FORBIDDEN] AddOn 'BuyEmAll' tried to call the protected function 'UseAction()'.
[string "@!BugGrabber/BugGrabber.lua"]:485: in function <!BugGrabber/BugGrabber.lua:485>
[string "=[C]"]: in function `UseAction'
[string "@Blizzard_FrameXML/Mainline/SecureTemplates.lua"]:342: in function `handler'
[string "@Blizzard_FrameXML/Mainline/SecureTemplates.lua"]:667: in function <...ddOns/Blizzard_FrameXML/Mainline/SecureTemplates.lua:649>
[string "@Blizzard_FrameXML/Mainline/SecureTemplates.lua"]:681: in function <...ddOns/Blizzard_FrameXML/Mainline/SecureTemplates.lua:674>
[string "@Blizzard_FrameXML/Mainline/SecureTemplates.lua"]:723: in function `SecureActionButton_OnClick'
[string "@Blizzard_ActionBar/Mainline/ActionButton.lua"]:100: in function `TryUseActionButton'
[string "@Blizzard_ActionBar/Mainline/ActionButton.lua"]:135: in function `ActionButtonDown'
[string "ACTIONBUTTON1"]:2: in function <[string "ACTIONBUTTON1"]:1>

Locals:
_ = Frame {
}
event = "ADDON_ACTION_FORBIDDEN"
events = <table> {
}

**Comment by raysmith59** [comment]
Here's the other error, pretty sure this also happened after a talent loadout change and then using a hotkey:

1x [ADDON_ACTION_BLOCKED] AddOn 'BuyEmAll' tried to call the protected function 'SpellFlyout:Hide()'.
[string "@!BugGrabber/BugGrabber.lua"]:485: in function <!BugGrabber/BugGrabber.lua:485>
[string "=[C]"]: in function `Hide'
[string "@Blizzard_FrameXML/Mainline/SecureTemplates.lua"]:341: in function `handler'
[string "@Blizzard_FrameXML/Mainline/SecureTemplates.lua"]:667: in function <...ddOns/Blizzard_FrameXML/Mainline/SecureTemplates.lua:649>
[string "@Blizzard_FrameXML/Mainline/SecureTemplates.lua"]:681: in function <...ddOns/Blizzard_FrameXML/Mainline/SecureTemplates.lua:674>
[string "@Blizzard_FrameXML/Mainline/SecureTemplates.lua"]:723: in function `SecureActionButton_OnClick'
[string "@Blizzard_ActionBar/Mainline/ActionButton.lua"]:100: in function `TryUseActionButton'
[string "@Blizzard_ActionBar/Mainline/ActionButton.lua"]:135: in function `ActionButtonDown'
[string "ACTIONBUTTON4"]:2: in function <[string "ACTIONBUTTON4"]:1>

Locals:
_ = Frame {
}
event = "ADDON_ACTION_BLOCKED"
events = <table> {
}

**Comment by raysmith59** [comment]
Decided to test immediately, ONLY BEA enabled, I don't have Bugsack enabled so I don't get the pretty error but I got this instead when I changed talent loadouts and then hit a hotkey:

![image](https://github.com/user-attachments/assets/944891b0-a556-4c60-877c-de37a8865861)

**Comment by kaelonR** [comment]
Should hopefully be resolved with v4.1.1. Would you mind testing again on that version @raysmith59?

**Comment by raysmith59** [comment]
New version seems to resolve my issue so I'm good, thanks!
